LLVMOffloadArch should link libdl to fix undefined symbol 'dlsym' when linking
--- llvm.orig/lib/OffloadArch/offload-arch/CMakeLists.txt
+++ llvm/lib/OffloadArch/offload-arch/CMakeLists.txt
@@ -3,6 +3,7 @@ add_llvm_tool(offload-arch
   ${CMAKE_CURRENT_SOURCE_DIR}/offload-arch.cpp
   DEPENDS generated-table LLVMOffloadArch
 )
+target_link_libraries(LLVMOffloadArch PRIVATE ${CMAKE_DL_LIBS})
 target_link_libraries(offload-arch PRIVATE LLVMOffloadArch)
 
 if(CMAKE_HOST_UNIX)
